### ISSUE TITLE
System.InvalidOperationException on C# UWP client

### ISSUE THREAD

**Issue by Tristyn** [aspnet/SignalR#2648]
In a blank UWP app:

```
namespace App1
{
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            
            Init();
        }

        private async void Init()
        {
            await Task.Delay(4000);
            // Wait for server to listen

            var hub = new HubConnectionBuilder()
                .WithUrl("http://localhost:5000/api/v1/myhub")
                .Build();

            await hub.StartAsync();
        }
    }
}
```

I get the exception `System.InvalidOperationException: 'Unable to connect to the server with any of the available transports.'`

I reproduced it at [https://github.com/Tristyn/SignalrUwpRepro](https://github.com/Tristyn/SignalrUwpRepro), by starting multiple projects in VS it will work on the console app, however throws on UWP.

I used Kestrel with https disabled for simplicity.

**Comment by BrennanConroy** [comment]
Please follow the diagnostics guide and gather the logs for the UWP app, https://github.com/aspnet/SignalR/wiki/Diagnostics-Guide#net-client-logging

**Comment by Tristyn** [comment]
Here is the debug log.

[debug log.txt](https://github.com/aspnet/SignalR/files/2202572/debug.log.txt)

The log states `System.IO.FileLoadException: Could not load file or assembly 'System.Threading.Tasks.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)` Adding the nuget package System.Threading.Tasks.Extensions v4.5.1 fixed the issue for me, I think you guys should be pulling that package?

**Comment by BrennanConroy** [comment]
We have https://github.com/aspnet/SignalR/issues/2474 tracking that

**Comment by Tristyn** [comment]
Thanks again. The internal FileLoadException (or an AggregateException) should be an inner exception to the InvalidOperationException to help users self diagnose. My coworker recalled a similar missing package on Signalr/Xamarin but the FileLoadException was wrapped which made it an quick temp fix

**Comment by Bunhy** [comment]
I also ran into this problem. And spent about 3 hours looking for the solution until I found this. Yes, using SignalR in UPW apps seems to require manual adding System.Threading.Tasks.Extensions nuget package. Or you get the misleading "Unable to connect to the server with any of the available transports" Exception. 

As Tristyn has already written: Shouldn't this be a dependency of SignalR.Client when installig in an UWP app?

**Comment by anurse** [comment]
See our [announcement regarding this topic](https://github.com/aspnet/Announcements/issues/306)

> Shouldn't this be a dependency of SignalR.Client when installig in an UWP app?

The problem isn't that we don't have the dependency, it's that we have a dependency on an earlier version that fails to load in UWP. This was an issue in the RTM version of .NET Core 2.1 which was resolved in a patch. Our practice is to depend upon the lowest patch version corresponding to the minor release we depend on and that version has this issue in UWP. Users are able to upgrade the package manually to resolve the issue but our policy is not to force this upgrade on all users in a patch release, so we haven't updated that reference yet.

https://github.com/aspnet/SignalR/issues/3209 is tracking fixing this in 2.2, when we have a chance to update to the next minor version of .NET Core's packages.

**Comment by anurse** [comment]
This is resolved in 2.2. The latest preview build (`1.1.0-preview3-35497`) of `Microsoft.AspNetCore.SignalR.Client` should work fine on UWP with no additional dependencies.
